charts/redpanda/chart_template_test.go

```diff
@@ -133,6 +137,39 @@ func TestTemplate(t *testing.T) {
 	}
 }
 
+func AssertSuperUsersAreValid(t *testing.T, manifests []byte) {
```

Comment:
A static assertions might be better here? I don't think this would catch invalid formatting like `user:%s(nil)`. It does catch the existing issue though because of the switch.